packages/frontend/src/lib/pnl.ts

```diff
@@ -70,25 +82,47 @@ export async function calcETHCollateralPnl(
           )
         }
         return acc
-      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO }))
-    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO }
+      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO, priceError: false }))
+    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO, priceError: true }
 
-  return currentVaultEthBalance.times(uniswapEthPrice).minus(deposits.minus(withdrawals))
+  return !priceError ? currentVaultEthBalance.times(uniswapEthPrice).minus(deposits.minus(withdrawals)) : BIG_ZERO
+}
+
+const getRelevantSwaps = (squeethAmount: BigNumber, swaps: swaps_swaps[], isWethToken0: boolean, isLong = false) => {
```

Comment:
what is defined to be a 'relevant swap'? I personally don't understand the logic of what we're defining as relevant here? Also might be good to add this as a comment describing the function

Reply:
Hmm still don't fully understand this - what lines are specifically getting swaps related to a user's current position? Where is the logic that defines what swap is related to a users current position vs. not?

Reply:
Starting from the most recent swap.
Line 105 - is adding squeeth amount of current swap to a variable "totalSqueeth"
Line 106 - is pushing the swap into the relevantSwaps array
Line 109 - is checking if the accumulated squeeth adds up to the user position squeeth amount.
Line 110 - is stoping the loop if condition on line 109 is true.
Line 113 - is returning the array with the swaps that got pushed into the array

Reply:
Ok I guess I don't understand how this is different than just using the swaps array? Seems like we are taking the swaps array, pushing all swaps into the relevant swaps array(?), and then what is the user position squeeth amount, like where is that coming from and how is that different than the accumulated squeeth here?

Reply:
Not sure if this will work as expected, i.e. if users buy and then mint oSQTH, isLong will end up being true, and will cause the squeethAmt of mint position seems to be added up as totalSqueeth in line 96&97

Reply:
@yichiehliu minted squeeth is not included in the swaps or squeethAmount. Minted squeeth is calculated `oSqueethBal.minus(squeethAmount)` in the code.
@alexisgauba Okay the user's squeeth is passed in as an argument from the positions context. Some of the swaps in the swaps array may be from previous positions that have been completely closed, so this ensures that we don't include them.

Reply:
@EstherSeyi Actually it is, cuz we are getting all the swaps data from a user, which includes mint/burn txs

Reply:
Ok, I think I was just confused myself with direct selling on uniswap